hoomd/HOOMDMath.cc

```diff
@@ -104,50 +104,6 @@ void export_hoomd_math_functions(pybind11::module& m)
             "w",
             [](const float4& s) { return (float)s.w; },
             [](float4& s, float v) { s.w = v; });
-#endif
-
-    pybind11::class_<Scalar2, std::shared_ptr<Scalar2>>(m, "Scalar2")
-        .def(pybind11::init<>())
-        .def_property(
-            "x",
-            [](const Scalar2& s) { return (Scalar)s.x; },
-            [](Scalar2& s, Scalar v) { s.x = v; })
-        .def_property(
-            "y",
-            [](const Scalar2& s) { return (Scalar)s.y; },
-            [](Scalar2& s, Scalar v) { s.y = v; });
-    pybind11::class_<Scalar3, std::shared_ptr<Scalar3>>(m, "Scalar3")
```

Comment:
How can we delete this and `make_scalar3` works fine?

Reply:
Because both `float3` and `double3` are exported. `make_scalar3` takes one of those or the other as an argument, depending on the value of `HOOMD_LONGREAL_SIZE` at build time. `pybind11` resolves typedefs to their actual type and finds that whatever `Scalar3` has been typedefed to exists.